frontend/src/Buildings/index.jsx

```diff
@@ -68,16 +67,29 @@ function Buildings() {
     setActiveId(id);
   }
 
-  function addNewBuilding(type) {
-    dispatch(addBuildingSuccess(type));
-  }
-
-  if (error) {
-    return <ErrorPopup message={`Oops, ${error.message}`} />;
+  async function addNewBuilding(type) {
+    const { type: actionType, payload } = await dispatch(addBuilding(type, token));
+    if (actionType === ADD_BUILDING_FAILURE) {
+      Toast.show({
+        type: 'warning',
+        duration: 3000,
+        text: payload,
+        buttonText: 'Okay',
+      });
+    } else {
+      Toast.show({
+        type: 'success',
+        duration: 3000,
+        text: 'Building is succesfully added',
+        buttonText: 'Okay',
+      });
+    }
   }
 
-  if (isLoading) {
-    return <ActivityIndicator size="large" color={Colors.tealColor} />;
+  if (isLoading || !buildings) {
```

Comment:
not sure why did you add `!buildings`?

Reply:
We had this before just for getBuilding part, I think I forget to delete it.